modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/ImageConvertWorkflowOperationHandler.java

```diff
@@ -0,0 +1,275 @@
+          String targetFileName = PathSupport.toSafeName(FilenameUtils.getName(targetElement.getURI().getPath()));
+          URI newTargetElementUri = workspace.moveTo(targetElement.getURI(), mediaPackage.getIdentifier().compact(),
+                  targetElement.getIdentifier(), targetFileName);
+          targetElement.setURI(newTargetElementUri);
+          targetElement.setChecksum(null);
+
+          // set flavor on target element
+          if (targetFlavor != null) {
+            targetElement.setFlavor(targetFlavor);
+            if (StringUtils.equalsAny("*", targetFlavor.getType())) {
+              targetElement.setFlavor(MediaPackageElementFlavor.flavor(
+                      sourceElement.getFlavor().getType(), targetElement.getFlavor().getSubtype()));
+            }
+            if (StringUtils.equalsAny("*", targetFlavor.getSubtype())) {
+              targetElement.setFlavor(MediaPackageElementFlavor.flavor(
+                      targetElement.getFlavor().getType(), sourceElement.getFlavor().getSubtype()));
+            }
+          }
+          // set tags on target element
+          List<String> fixedTags = new ArrayList<>();
+          List<String> additionalTags = new ArrayList<>();
+          List<String> removingTags = new ArrayList<>();
+          for (String targetTag : asList(targetTagsOption)) {
+            if (!StringUtils.startsWithAny(targetTag, "+", "-")) {
+              if (additionalTags.size() > 0 || removingTags.size() > 0) {
+                logger.warn("You may not mix fixed tags and tag changes. "
+                        + "Please review target-tags option on image-convert operation of your workflow definition. "
+                        + "The tag {} is not prefixed with '+' or '-'.", targetTag);
+              }
+              fixedTags.add(targetTag);
+            } else if (StringUtils.startsWith(targetTag, "+")) {
+              additionalTags.add(StringUtils.substring(targetTag, 1));
+            } else if (StringUtils.startsWith(targetTag, "-")) {
+              removingTags.add(StringUtils.substring(targetTag, 1));
+            }
+          }
+          targetElement.clearTags();
+          if (fixedTags.isEmpty() && (!additionalTags.isEmpty() || !removingTags.isEmpty())) {
+            for (String tag : sourceElement.getTags()) {
+              targetElement.addTag(tag);
+            }
+          }
+          for (String targetTag : fixedTags) {
+            targetElement.addTag(targetTag);
+          }
+          for (String additionalTag : additionalTags) {
+            targetElement.addTag(additionalTag);
+          }
+          for (String removingTag : removingTags) {
+            targetElement.removeTag(removingTag);
+          }
+          mediaPackage.addDerived(targetElement, sourceElement);
+        }
+      }
+      return createResult(mediaPackage, WorkflowOperationResult.Action.CONTINUE);
+    } catch (WorkflowOperationException ex) {
+      throw ex;
```

Comment:
Why catch this to simply throw it again?

Reply:
I missed that you are using this as an exception to catching `Throwable`. That makes sense. Though I'm wondering if catching `Throwable` does. Does it make sense to convert system `Error` to `Exception`?